spark/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java

```diff
@@ -423,7 +423,11 @@ public int getProgress(InterpreterContext context) {
     }
 
     synchronized (statementFinishedNotifier) {
-      while (statementOutput == null) {
+      long startTime = System.currentTimeMillis();
+      while (statementOutput == null
```

Comment:
I think we should at least log it when it times out.
I don't know how we handle error in completion() but maybe a `InterpreterResult(Code.ERROR)`?